### DISCUSSION TITLE
How do I get the renderer to work?

### DISCUSSION THREAD

**Discussion by dhfreinkel** [codefori/vscode-ibmi#795] [IBM i]
I have the renderer installed but cannot see how to use it. Tried to find a video and found the one Liam created but that is too complicated and does not look correct just to be able to run the renderer.

TIA.

**Comment by Wright4i** [comment]
Open a dspf or prtf and find the Preview links embedded in the source window. That will open up the renderer.

**Reply by dhfreinkel** [reply]
I cannot find any preview links in the source. Not sure I know what these links look like. I scanned the menu's and various options but do not see anything.

**Reply by worksofliam** [reply]
@dhfreinkel Be sure that the language id is set the `dds.dspf` (in the bottom right of VS Code). Maybe share a screenshot when you have a display file open next. That should tell us what the issue is.

**Reply by m-tyler** [reply]
@dhfreinkel this is what a DSPF with preview looks like in code
![image](https://user-images.githubusercontent.com/1353665/186729895-c4dabb8d-269c-4a2c-aeed-433986c88d63.png)
 There's a bonus link for previewing all formats at the same time.  
![image](https://user-images.githubusercontent.com/1353665/186730122-1ca01bb8-67c9-4978-a338-cee8b394a4e4.png)


The preview tool also works wonderfully for PRTFs. 

If you open the preview, drag it to the right until it becomes a new tab group.  Press Ctrl+B to minimize the primary sidebar view.  You can then see the changes you make in the source appear on the preview within a few seconds, depending on the settings you have.
